src/Cache.php

```diff
@@ -0,0 +1,102 @@
+<?php
+declare(strict_types=1);
+
+namespace customiesdevs\customies\util;
+
+use pocketmine\utils\Filesystem;
+use pocketmine\utils\SingletonTrait;
+use function asort;
+use function file_exists;
+use function file_get_contents;
+use function gzcompress;
+use function gzuncompress;
+use function igbinary_serialize;
+use function igbinary_unserialize;
+
+final class Cache {
+    use SingletonTrait;
+
+	private int $nextBlockID;
+	private int $nextItemID;
+	private string $file;
+
+	/**
+	 * @var array
+	 */
+	private array $itemCache = [];
+
+	/**
+	 * @var array
+	 */
+	private array $blockCache = [];
+
+	public function __construct(int $startingBlockID, int $startingItemID, string $cacheFile) {
```

Comment:
Is there any reason to have the starting id arguments here? Are we considering making them configurable at some point?

Reply:
removed these